### DISCUSSION TITLE
VersionPress 4.0 support

### DISCUSSION THREAD

**Discussion by Ramoonus** [pods-framework/pods#6747] [Ideas]
https://blog.versionpress.net/2017/07/versionpress-4-0-beta/

**Comment by borekb** [comment]
Cool to see this ticket! Some previous discussion:

- https://github.com/versionpress/versionpress/issues/1036#issuecomment-217214337
- https://github.com/versionpress/versionpress/issues/1036#issuecomment-217222843

**Comment by sc0ttkclark** [comment]
I think a good first goal is to ensure versioning the Pods configurations themselves works correctly. A main issue with versioning the configs themselves are dealing with table schema changes, field renaming (meta keys won't match, etc), field deletions (meta values will still be there, table columns/values will still be there).

**Comment by Ramoonus** [comment]
it will be tough, but a huge functionality improvement

**Comment by lougreenwood** [comment]
FYI - I'm already storing dumps of my entire WP dev installation in git and dumping changes into the dump in branches. So may be able to give some insight into handling Pods config changes via git...

**Comment by pavelevap** [comment]
I am using Pods for my projects, so I started to play with VersionPress support.

- Pods configuration simply works, post types and postmeta are supported by default, only entities have to be checked. I tested adding several fields and I did not notice any problem.
- I also added support for `podsrel` table and related entities to start tracking data and it seems that it also works, but I will test also renaming fields, etc (PR incoming).

![image](https://user-images.githubusercontent.com/3008007/28626145-818267a2-721e-11e7-87c4-b16ee7df09f2.png)

- Shortcode support has to be added and also some advanced components will be challenging (especially custom table storage).

Some related questions:
- Relationships and postmeta are deleted and then added (not updated) by default?
- For relationship field with name `test-relation-field` there are 2 similar postmeta with ID values `_pods_test-relation-field` (serialized value) and `test-relation-field` (simple value). I am not sure about the second one (there is no prefix), is this postmeta important?
- I tried to rename some field and it seems to me that it can be renamed only once? Postmeta `old_name` is changed, but when I change it once again, there is no relation to previous name (and that is why mentioned postmeta `test-relation-field` will not work correctly)?

**Comment by lougreenwood** [comment]
Nice work @pavelevap! 👍
